src/main/java/net/fabricmc/api/EnvironmentInterface.java

```diff
@@ -0,0 +1,10 @@
+package net.fabricmc.api;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+@Retention(RetentionPolicy.CLASS)
+public @interface EnvironmentInterface {
+	EnvType value();
+	Class<?>[] itf();
```

Comment:
You sure we don't need to use a String here to avoid reflection issues?

Reply:
`RetentionPolicy.CLASS` should make the annotation invisible at runtime, so reflecting on the class should not produce the annotation and thus not load anything.

However, if the jar merger ignores the retention policy of `@EnvironmentInterface` like it does with `@Environment` (which is CLASS by default) by producing visible annotations, I believe there will be issues.